src/modules/endpoint/ContentLoadCriterion.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import moment from 'moment-timezone'
+import Moment from 'moment'
+
+const MAX_CONTENT_AGE = 24
+
+export type ContentLoadCriterionType = {|
+  peek?: boolean,
+  forceUpdate: boolean,
+  shouldRefreshResources: boolean
+|}
+
+export class ContentLoadCriterion {
+  _forceUpdate: boolean
+  _shouldRefreshResources: boolean
+  _peek: boolean
+
+  constructor ({peek = false, forceUpdate, shouldRefreshResources}: ContentLoadCriterionType) {
+    this._peek = peek
+    this._forceUpdate = forceUpdate
+    this._shouldRefreshResources = shouldRefreshResources
+  }
+
+  shouldUpdate (lastUpdate: ?Moment): boolean {
+    // The last update was more than 24h ago or a refresh should be forced
+    return this._forceUpdate || !lastUpdate ||
+      lastUpdate.isBefore(moment.tz('UTC').subtract(MAX_CONTENT_AGE, 'hours'))
+  }
+
+  peek (): boolean {
+    return this._peek
+  }
+
+  shouldRefreshResources (): boolean {
+    return this._shouldRefreshResources || this._peek
```

Comment:
Why should we refresh resources if we're peeking? I'd say should explicitly *not* do it, because we won't need most of it?